### DISCUSSION TITLE
using external module  in seed function

### DISCUSSION THREAD

**Discussion by wvhulle** [prisma/prisma#15240] [Q&A] [answered]
I have configured a command to run the seed function and put part of the functionality in a separate module. However, when I try to import the module I get the following error? Why can I not use `import { function } from './util`?

```
Running seed command ` node --loader ts-node/esm src/lib/data/seed.ts` ...
(node:2925891) ExperimentalWarning: Custom ESM Loaders is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
/home/willem/Documenten/dsp/node_modules/.pnpm/ts-node@10.9.1_rezidhttjovwjanvxmgta63sl4/node_modules/ts-node/dist-raw/node-internal-modules-esm-resolve.js:366
    throw new ERR_MODULE_NOT_FOUND(
          ^
CustomError: Cannot find module '/home/willem/Documenten/dsp/src/lib/data/util' imported from /home/willem/Documenten/dsp/src/lib/data/seed.ts
```

**Comment by ludralph** [comment, marked as answer]
Hi @wvhulle 👋 

The "Error [ERR_MODULE_NOT_FOUND]: Cannot find module" occurs when you set the type attribute to module in your package.json file, but omit the file extension when importing. To solve the error, specify the extension when importing local files.

The package.json file has the type attribute set to module.
package.json
```js
{
  type: module
// ... rest
}
```

To solve the error, make sure to include the extension when importing local files with type set to module in your Node.js project.
 import { function } from './util.js

The [node docs](https://nodejs.org/api/esm.html#mandatory-file-extensions) state that a file extension must be provided when using the import keyword to resolve relative or absolute specifiers.

This behavior matches how import works in browser environments.

Directory specifiers like ./my-folder/my-file.js must also be fully specified.
